Core/include/Acts/TrackFitting/detail/GsfActor.hpp

```diff
@@ -32,6 +32,23 @@
 namespace Acts {
 namespace detail {
 
+/// Small Helper class that allows to carry a tmporary value until we decide to
+/// update the actual value. The temporary value is deliberatly only accessable
+/// with a mutual reference
+template <typename T>
+class Updatable {
```

Comment:
I have to admit I don't fully understand what this class is for. Is it double "buffering" the value so you can go back to the original one? 

I don't think I see where it would get reset. 

Why the separate values then?

Reply:
Yeah so not too mutch behind this:

The use case is the following:
I have a variable that gets updated constantly in the propagator.

But I'm only interested into the updated value up to a certain point (e.g., the last measurement).
But I don't know when this will come, so I need to keep both the value at the last measurement surface and the current one.

Of course this is easily doable with two simple variables, but I didnt want to have many `tmpXYZ` and `lastValidXYZ` but rather encapsulate this...